From 4ed9e6630a9f3cd08296c0dc5ec06cd6a58ccb0b Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Wed, 6 Aug 2014 22:34:25 +0800
Subject: [PATCH 4748/5965] PD #93541:Remove unused function type warning

Change-Id: Icb184db2c1079878a30c78a405008f725fd94e8b

Conflicts:

	arch/arm/mach-meson8/vpu.c

Conflicts:

	drivers/amlogic/amports/vh265.c
---
 Makefile                                      |  3 ++-
 arch/arm/kernel/process.c                     |  4 ++--
 arch/arm/mach-meson8/gpiotest.c               |  4 ++--
 arch/arm/mach-meson8/vpu.c                    |  1 -
 arch/arm/mach-meson8b/vpu.c                   |  4 ++--
 arch/arm/mm/init.c                            |  3 ++-
 drivers/amlogic/amports/rdma.c                |  3 ++-
 drivers/amlogic/amports/vh265.c               |  1 -
 drivers/amlogic/camera/hm2057.c               |  5 ++++-
 drivers/amlogic/camera/sp0a19.c               |  4 ++--
 drivers/amlogic/camera/sp1628.c               |  4 ++--
 .../backlight/aml_bl_extern/i2c_lp8556.c      |  4 ++--
 drivers/amlogic/ethernet/am_net8218.c         |  7 +++++-
 drivers/amlogic/ethernet/phy/am_rtl8211f.c    |  4 ++--
 .../ext_hdmiin/sii9233/__sii9233/infofrm.c    |  4 +++-
 .../amlogic/ext_hdmiin/sii9233/sii9233_drv.c  |  4 ++--
 drivers/amlogic/input/new_remote/irblaster.c  |  3 ++-
 drivers/amlogic/input/sensor/lsm303d.c        |  9 ++++----
 drivers/amlogic/input/sensor/mir3da_core.c    |  3 ++-
 drivers/amlogic/input/sensor/mir3da_cust.c    |  4 ++--
 .../touchscreen/gslx680/amlogic_gslX680.c     |  4 ++--
 drivers/amlogic/ionvideo/ionvideo.c           |  4 ++--
 drivers/amlogic/secure/aml_aes/aml_aes.c      |  4 ++--
 drivers/amlogic/spi_nor/apollo_spi_nor.c      |  6 ++---
 include/linux/thermal.h                       | 10 ---------
 sound/soc/aml/m6/aml_pcm.c                    | 22 +++++++++++--------
 sound/soc/codecs/aml_pmu3.c                   |  4 ++--
 27 files changed, 69 insertions(+), 63 deletions(-)

diff --git a/Makefile b/Makefile
index dd5a692edae3..54c82f6994aa 100755
--- a/Makefile
+++ b/Makefile
@@ -381,7 +381,8 @@ KBUILD_CFLAGS   += -Werror=enum-compare \
 		   -Werror=unused-value \
 		   -Werror=maybe-uninitialized \
 		   -Werror=unused-variable \
-		   -Werror=format
+		   -Werror=format \
+		   -Werror=unused-function
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
 KBUILD_AFLAGS   := -D__ASSEMBLY__
diff --git a/arch/arm/kernel/process.c b/arch/arm/kernel/process.c
index c9048a195ebc..82efbf9f546e 100755
--- a/arch/arm/kernel/process.c
+++ b/arch/arm/kernel/process.c
@@ -157,11 +157,11 @@ void soft_restart(unsigned long addr)
 	/* Should never get here. */
 	BUG();
 }
-
+#if 0
 static void null_restart(char mode, const char *cmd)
 {
 }
-
+#endif
 void arm_machine_restart(char mode, const char *cmd)
 {
 #if defined (CONFIG_PLAT_MESON) && !defined(CONFIG_ARCH_MESON2)
diff --git a/arch/arm/mach-meson8/gpiotest.c b/arch/arm/mach-meson8/gpiotest.c
index 953f224cad27..03fb3229748f 100755
--- a/arch/arm/mach-meson8/gpiotest.c
+++ b/arch/arm/mach-meson8/gpiotest.c
@@ -252,7 +252,7 @@ static void set_gpio_pinmux(struct gpio_group_info_s* g_info, unsigned mask)
 	}
 		
 }
-
+#if 0
 static void free_gpio_pinmux(struct gpio_group_info_s* g_info, unsigned mask)
 {
 	int i;
@@ -262,7 +262,7 @@ static void free_gpio_pinmux(struct gpio_group_info_s* g_info, unsigned mask)
 	}
 		
 }
-
+#endif
 static void revert_gpiotest_pinmux(void)
 {
 	int i;
diff --git a/arch/arm/mach-meson8/vpu.c b/arch/arm/mach-meson8/vpu.c
index 479ac638dc28..17115438764a 100755
--- a/arch/arm/mach-meson8/vpu.c
+++ b/arch/arm/mach-meson8/vpu.c
@@ -754,7 +754,6 @@ static void vpu_driver_init(void)
     //Remove VPU_HDMI ISO
     aml_set_reg32_bits(P_AO_RTI_GEN_PWR_SLEEP0, 0, 9, 1); // [9] VPU_HDMI
 }
-
 static void vpu_driver_disable(void)
 {
     vpu_config.mem_pd0 = aml_read_reg32(P_HHI_VPU_MEM_PD_REG0);
diff --git a/arch/arm/mach-meson8b/vpu.c b/arch/arm/mach-meson8b/vpu.c
index 97956058e803..a1908f7c8ff9 100755
--- a/arch/arm/mach-meson8b/vpu.c
+++ b/arch/arm/mach-meson8b/vpu.c
@@ -636,7 +636,7 @@ static struct class aml_vpu_debug_class = {
 	.class_attrs = vpu_debug_class_attrs,
 };
 //*********************************************************//
-
+#if 0
 static void vpu_driver_init(void)
 {	
 	set_vpu_clk(vpu_config.clk_level);
@@ -656,7 +656,7 @@ static void vpu_driver_disable(void)
 	
 	aml_set_reg32_bits(P_HHI_VPU_CLK_CNTL, 0, 8, 1);
 }
-
+#endif
 #ifdef CONFIG_PM
 static int vpu_suspend(struct platform_device *pdev, pm_message_t state)
 {
diff --git a/arch/arm/mm/init.c b/arch/arm/mm/init.c
index 88b349d254d6..1ab66303b016 100644
--- a/arch/arm/mm/init.c
+++ b/arch/arm/mm/init.c
@@ -468,6 +468,7 @@ free_memmap(unsigned long start_pfn, unsigned long end_pfn)
 /*
  * The mem_map array can get very big.  Free the unused area of the memory map.
  */
+#ifndef CONFIG_PLAT_MESON
 static void __init free_unused_memmap(struct meminfo *mi)
 {
 	unsigned long bank_start, prev_bank_end = 0;
@@ -518,7 +519,7 @@ static void __init free_unused_memmap(struct meminfo *mi)
 			    ALIGN(prev_bank_end, PAGES_PER_SECTION));
 #endif
 }
-
+#endif
 #ifdef CONFIG_HIGHMEM
 static inline void free_area_high(unsigned long pfn, unsigned long end)
 {
diff --git a/drivers/amlogic/amports/rdma.c b/drivers/amlogic/amports/rdma.c
index 8ce01435d1e1..7767f35c58d5 100755
--- a/drivers/amlogic/amports/rdma.c
+++ b/drivers/amlogic/amports/rdma.c
@@ -340,6 +340,7 @@ void vsync_rdma_config_pre(void)
 }    
 EXPORT_SYMBOL(vsync_rdma_config_pre);
 
+#if MESON_CPU_TYPE < MESON_CPU_TYPE_MESON8
 static irqreturn_t rdma_isr(int irq, void *dev_id)
 {
 #ifdef CONFIG_RDMA_IN_RDMAIRQ
@@ -388,7 +389,7 @@ static irqreturn_t rdma_isr(int irq, void *dev_id)
 #endif    
     return IRQ_HANDLED;
 }
-
+#endif
 static int __init rmda_early_init(void)
 {
     ulong rmda_table_addr;
diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index 8e39f34c2395..65943d3d49fd 100755
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -1126,7 +1126,6 @@ static void dump_pic_list(hevc_stru_t* hevc)
 		pic = pic->next;
 	}
 }
-
 static PIC_t* output_pic(hevc_stru_t* hevc, unsigned char flush_flag)
 {
 	int num_pic_not_yet_display = 0;
diff --git a/drivers/amlogic/camera/hm2057.c b/drivers/amlogic/camera/hm2057.c
index 29b1733f3d64..f2ef381e14ef 100755
--- a/drivers/amlogic/camera/hm2057.c
+++ b/drivers/amlogic/camera/hm2057.c
@@ -1485,6 +1485,7 @@ void hm2057_set_param_banding(struct hm2057_device *dev,enum  camera_banding_fli
 		
 				}
 }
+#if 0
 static int set_flip(struct hm2057_device *dev)
 {
    	struct i2c_client *client = v4l2_get_subdevdata(&dev->sd);
@@ -1501,7 +1502,7 @@ static int set_flip(struct hm2057_device *dev)
         }
         return 0;
 }
-
+#endif
 
 
 void hm2057_set_resolution(struct hm2057_device *dev,int height,int width)
@@ -1702,6 +1703,7 @@ static void power_down_hm2057(struct hm2057_device *dev)
 #define TSTAMP_MAX_Y	(TSTAMP_MIN_Y + 15)
 #define TSTAMP_INPUT_X	10
 #define TSTAMP_MIN_X	(54 + TSTAMP_INPUT_X)
+#if 0
 static int getCameraState(){
     int i=get_adc_sample(0);
 	if(i>374&&i<391){
@@ -1710,6 +1712,7 @@ static int getCameraState(){
         return 0;
 	}
 }
+#endif
 static void hm2057_fillbuff(struct hm2057_fh *fh, struct hm2057_buffer *buf)
 {
 	//printk("----------- %s \n",__func__);
diff --git a/drivers/amlogic/camera/sp0a19.c b/drivers/amlogic/camera/sp0a19.c
index a61bb457334a..531cf5f63fca 100755
--- a/drivers/amlogic/camera/sp0a19.c
+++ b/drivers/amlogic/camera/sp0a19.c
@@ -1995,7 +1995,7 @@ static int sp0a19_setting(struct sp0a19_device *dev,int PROP_ID,int value )
 	return ret;
 
 }
-
+#if 0
 static void power_down_sp0a19(struct sp0a19_device *dev)
 {
 	struct i2c_client *client = v4l2_get_subdevdata(&dev->sd);
@@ -2010,7 +2010,7 @@ static void power_down_sp0a19(struct sp0a19_device *dev)
 	msleep(5);
 	return;    
 }
-
+#endif
 /* ------------------------------------------------------------------
 	DMA and thread functions
    ------------------------------------------------------------------*/
diff --git a/drivers/amlogic/camera/sp1628.c b/drivers/amlogic/camera/sp1628.c
index 9522bfacddda..de040a17759b 100755
--- a/drivers/amlogic/camera/sp1628.c
+++ b/drivers/amlogic/camera/sp1628.c
@@ -964,7 +964,7 @@ void SP1628_init_regs(struct sp1628_device *dev)
 
 #endif
 
-
+#if 0
 static int set_flip(struct sp1628_device *dev)
 {
 	struct i2c_client *client = v4l2_get_subdevdata(&dev->sd);
@@ -982,7 +982,7 @@ static int set_flip(struct sp1628_device *dev)
         }
         return 0;
 }
-
+#endif
 
 static void sp1628_set_resolution(struct sp1628_device *dev,int height,int width)
 {
diff --git a/drivers/amlogic/display/backlight/aml_bl_extern/i2c_lp8556.c b/drivers/amlogic/display/backlight/aml_bl_extern/i2c_lp8556.c
index 65b7e4a6b406..d8b048ff0378 100755
--- a/drivers/amlogic/display/backlight/aml_bl_extern/i2c_lp8556.c
+++ b/drivers/amlogic/display/backlight/aml_bl_extern/i2c_lp8556.c
@@ -71,7 +71,7 @@ static int aml_i2c_write(struct i2c_client *i2client,unsigned char *buff, unsign
     
     return res;
 }
-
+#if 0
 static int aml_i2c_read(struct i2c_client *i2client,unsigned char *buff, unsigned len)
 {
     int res = 0;
@@ -96,7 +96,7 @@ static int aml_i2c_read(struct i2c_client *i2client,unsigned char *buff, unsigne
 
     return res;
 }
-
+#endif
 static int bl_extern_set_level(unsigned int level)
 {
     unsigned char tData[3];
diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 358924362f26..52d93aafcea0 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -1890,9 +1890,11 @@ M8baby
 G9TV
  28
 */
+#if 0
 static unsigned int get_cpuid(){
 	return READ_CBUS_REG(0x1f53)&0xff;
 }
+#endif
 /* --------------------------------------------------------------------------*/
 /**
  * @brief  probe_init
@@ -3013,13 +3015,14 @@ static int ethernet_remove(struct platform_device *pdev)
  * @return
  */
 /* --------------------------------------------------------------------------*/
+#ifndef  CONFIG_PM
 static int ethernet_suspend(struct platform_device *dev, pm_message_t event)
 {
 	printk("ethernet_suspend!\n");
 	netdev_close(my_ndev);	
 	return 0;
 }
-
+#endif
 
 /* --------------------------------------------------------------------------*/
 /**
@@ -3030,6 +3033,7 @@ static int ethernet_suspend(struct platform_device *dev, pm_message_t event)
  * @return
  */
 /* --------------------------------------------------------------------------*/
+#if 0
 static int ethernet_resume(struct platform_device *dev)
 {
 	int res = 0;
@@ -3042,6 +3046,7 @@ static int ethernet_resume(struct platform_device *dev)
 
 	return 0;
 }
+#endif
 #ifdef CONFIG_OF
 static const struct of_device_id eth_dt_match[]={
 	{	.compatible 	= "amlogic,meson-eth",
diff --git a/drivers/amlogic/ethernet/phy/am_rtl8211f.c b/drivers/amlogic/ethernet/phy/am_rtl8211f.c
index a7156e6b6ad3..d64b1af6c24d 100755
--- a/drivers/amlogic/ethernet/phy/am_rtl8211f.c
+++ b/drivers/amlogic/ethernet/phy/am_rtl8211f.c
@@ -15,7 +15,7 @@
 MODULE_DESCRIPTION("Realtek PHY driver");
 MODULE_AUTHOR("Johnson Leung");
 MODULE_LICENSE("GPL");
-
+#if 0
 static int rtl821x_ack_interrupt(struct phy_device *phydev)
 {
 	int err;
@@ -36,7 +36,7 @@ static int rtl8211e_config_intr(struct phy_device *phydev)
 
 	return err;
 }
-
+#endif
 static int rtl8211e_config_init(struct phy_device *phydev)
 {
 	int val;
diff --git a/drivers/amlogic/ext_hdmiin/sii9233/__sii9233/infofrm.c b/drivers/amlogic/ext_hdmiin/sii9233/__sii9233/infofrm.c
index 98eeb7cc4010..cc298f3075ae 100755
--- a/drivers/amlogic/ext_hdmiin/sii9233/__sii9233/infofrm.c
+++ b/drivers/amlogic/ext_hdmiin/sii9233/__sii9233/infofrm.c
@@ -129,7 +129,7 @@ static IfCtrl_t IfCtrl;
 
 /***** local function prototypes *********************************************/
 
-static uint8_t GetIfTypeFromUnreqBuffer ( void );
+//static uint8_t GetIfTypeFromUnreqBuffer ( void );
 static void SetDecodeIfBuffers ( const DevAddr_t DevAddr, const uint8_t bDecodeIfType );
 static void SetIfTo ( SelectIf_t eSelectIf, const uint16_t wTo );
 static bool_t GetIf( const uint8_t bIfType );
@@ -152,9 +152,11 @@ static uint8_t GetIfTypeFromUnreqBuffer ( void ){
     return RegisterRead( REG__UNREQ_TYPE );
 }
 #endif
+#if (CONF__DEBUG_PRINT == ENABLE)
 static uint8_t GetMpegDecodeAddr ( void ){
     return RegisterRead( REG__MPEG_DECODE);
 }
+#endif
 /*****************************************************************************/
 /**
  *  The description of the function print_tabl().
diff --git a/drivers/amlogic/ext_hdmiin/sii9233/sii9233_drv.c b/drivers/amlogic/ext_hdmiin/sii9233/sii9233_drv.c
index 3f597db4da77..f72ec4c22b45 100755
--- a/drivers/amlogic/ext_hdmiin/sii9233/sii9233_drv.c
+++ b/drivers/amlogic/ext_hdmiin/sii9233/sii9233_drv.c
@@ -409,12 +409,11 @@ static ssize_t sii9233a_debug_store(struct class *class, struct class_attribute
 	kfree(p);
 	return count;
 }
-
+#if 0
 static ssize_t sii9233a_reset_show(struct class *class, struct class_attribute *attr, char *buf)
 {
 	return 0;
 }
-
 static ssize_t sii9233a_reset_store(struct class *class, struct class_attribute *attr,
 									const char *buf, size_t count)
 {
@@ -431,6 +430,7 @@ static ssize_t sii9233a_reset_store(struct class *class, struct class_attribute
     return count;
 }
 
+#endif
 static ssize_t sii9233a_port_show(struct class *class, struct class_attribute *attr, char *buf)
 {
 	char port = 0xff;
diff --git a/drivers/amlogic/input/new_remote/irblaster.c b/drivers/amlogic/input/new_remote/irblaster.c
index 014501198247..e68543fa2db9 100755
--- a/drivers/amlogic/input/new_remote/irblaster.c
+++ b/drivers/amlogic/input/new_remote/irblaster.c
@@ -91,7 +91,7 @@ static void send_all_frame(struct blaster_window * creat_window){
 	irblaster_dbg("The all frame finished !!\n");    
 
 }
-
+#if 0
 static void send_nec_frame(struct blaster_window * creat_window){
 	int i;
 	// set init_high valid and enable the ir_blaster
@@ -256,6 +256,7 @@ static void send_duokan_frame(struct blaster_window *creat_window){
 
 }
 
+#endif
 
 static ssize_t dbg_operate(struct device * dev, struct device_attribute *attr, const char * buf, size_t count)
 {
diff --git a/drivers/amlogic/input/sensor/lsm303d.c b/drivers/amlogic/input/sensor/lsm303d.c
index fbaa30f93775..4fe10a1929b3 100644
--- a/drivers/amlogic/input/sensor/lsm303d.c
+++ b/drivers/amlogic/input/sensor/lsm303d.c
@@ -1827,7 +1827,7 @@ static int write_gen_int(struct lsm303d_status *stat,
 	atomic_set(&ie->enable, val);
 	return err;
 }
-
+#if 0
 static int write_duration_threshold_int(struct lsm303d_status *stat, 
 					struct interrupt_value *ie, int val) 
 {
@@ -1877,7 +1877,6 @@ static int write_threshold_mag_int(struct lsm303d_status *stat,
 
 	return err;
 }
-
 static ssize_t attr_get_gen1_status(struct kobject *kobj,
 						struct kobj_attribute *attr,
 						char *buf)
@@ -2157,7 +2156,8 @@ static ssize_t attr_set_gen_mag_threshold(struct kobject *kobj,
 	
 	return size;
 }
-
+#endif
+#if 0
 static int get_axis(struct lsm303d_status *stat, 
 					int generator, const char *name) {
 
@@ -2234,7 +2234,6 @@ static int set_axis(struct lsm303d_status *stat, int generator,
 
 	return err;
 }
-
 static ssize_t attr_get_gen1_axis(struct kobject *kobj,
 						struct kobj_attribute *attr,
 						char *buf)
@@ -2389,7 +2388,7 @@ static ssize_t attr_set_gen2_and_or(struct kobject *kobj,
 
 	return size;
 }
-
+#endif
 static ssize_t attr_set_pin_conf(struct device *dev, 
 						struct device_attribute *attr,
 						const char *buf, size_t size)
diff --git a/drivers/amlogic/input/sensor/mir3da_core.c b/drivers/amlogic/input/sensor/mir3da_core.c
index 28b02edf4645..fecad37f7e73 100755
--- a/drivers/amlogic/input/sensor/mir3da_core.c
+++ b/drivers/amlogic/input/sensor/mir3da_core.c
@@ -345,6 +345,7 @@ static char isZStick(void){
 #endif /* ! MIR3DA_STK_TEMP_SOLUTION */
 
 #if FILTER_AVERAGE_ENHANCE
+#if 0
 static short filter_average(short preAve, short sample, int paramN, int* refNum)
 {
  #if FILTER_AVERAGE_EX
@@ -363,7 +364,6 @@ static short filter_average(short preAve, short sample, int paramN, int* refNum)
 
     return preAve + (sample - preAve)/paramN;
 }
-
 static int filter_average_enhance(FilterAverageContext* fac, short sample)
 {
     if (fac == 0){
@@ -384,6 +384,7 @@ static int filter_average_enhance(FilterAverageContext* fac, short sample)
 
     return fac->sample_h;    
 }
+#endif
 #endif /* ! FILTER_AVERAGE_ENHANCE */
 
 int mir3da_register_read(MIR_HANDLE handle, short addr, unsigned char *data)
diff --git a/drivers/amlogic/input/sensor/mir3da_cust.c b/drivers/amlogic/input/sensor/mir3da_cust.c
index f9639c856298..8b2ab462b876 100755
--- a/drivers/amlogic/input/sensor/mir3da_cust.c
+++ b/drivers/amlogic/input/sensor/mir3da_cust.c
@@ -442,7 +442,7 @@ static ssize_t mir3da_axis_data_show(struct device *dev,
 
     return count;
 }
-
+#if 0
 static ssize_t mir3da_odr_show(struct device *dev,
                    struct device_attribute *attr, char *buf)
 {
@@ -458,7 +458,6 @@ static ssize_t mir3da_odr_show(struct device *dev,
     
     return ret;
 }
-
 static ssize_t mir3da_odr_store(struct device *dev,struct device_attribute *attr,const char *buf, size_t count)
 {
     int ret;
@@ -475,6 +474,7 @@ static ssize_t mir3da_odr_store(struct device *dev,struct device_attribute *attr
     
     return ret;
 }
+#endif
 
 static ssize_t mir3da_reg_data_store(struct device *dev,
            struct device_attribute *attr, const char *buf, size_t count)
diff --git a/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c b/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
index b615e0ae8319..4151c02a4cc3 100755
--- a/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
+++ b/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
@@ -240,7 +240,7 @@ static inline u16 join_bytes(u8 a, u8 b)
 	ab = ab << 8 | b;
 	return ab;
 }
-
+#if 0
 static u32 gsl_read_interface(struct i2c_client *client, u8 reg, u8 *buf, u32 num)
 {
 	struct i2c_msg xfer_msg[2];
@@ -262,7 +262,7 @@ static u32 gsl_read_interface(struct i2c_client *client, u8 reg, u8 *buf, u32 nu
 
 	return i2c_transfer(client->adapter, xfer_msg, ARRAY_SIZE(xfer_msg)) == ARRAY_SIZE(xfer_msg) ? 0 : -EFAULT;
 }
-
+#endif
 static u32 gsl_write_interface(struct i2c_client *client, const u8 reg, u8 *buf, u32 num)
 {
 	struct i2c_msg xfer_msg[1];
diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index d959d35b0892..6c21c7bfbf52 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -135,7 +135,7 @@ int is_ionvideo_active(void) {
     return is_actived;
 }
 EXPORT_SYMBOL(is_ionvideo_active);
-
+#if 0
 static void videoc_omx_compute_pts(struct ionvideo_dev *dev, struct vframe_s* vf) {
     if (vf->pts) {
         timestamp_vpts_set(vf->pts);
@@ -150,7 +150,7 @@ static void videoc_omx_compute_pts(struct ionvideo_dev *dev, struct vframe_s* vf
         dev->pts = timestamp_vpts_get();
     }
 }
-
+#endif
 static int ionvideo_fillbuff(struct ionvideo_dev *dev, struct ionvideo_buffer *buf) {
 
     struct vframe_s* vf;
diff --git a/drivers/amlogic/secure/aml_aes/aml_aes.c b/drivers/amlogic/secure/aml_aes/aml_aes.c
index 2979191b4ec6..f242a3892e9f 100644
--- a/drivers/amlogic/secure/aml_aes/aml_aes.c
+++ b/drivers/amlogic/secure/aml_aes/aml_aes.c
@@ -163,7 +163,7 @@ static int aes_algorithm_enc_core(uint8_t data[16],const aes_key_context_t* ks )
 	return 0;
 }
 
-
+#if 0
 static void aes_inv_shift_row(void* data, uint8_t shift){
 	uint8_t tmp[4];
 	tmp[0] = ((uint8_t*)data)[(4+0-shift)&3];
@@ -172,7 +172,7 @@ static void aes_inv_shift_row(void* data, uint8_t shift){
 	tmp[3] = ((uint8_t*)data)[(4+3-shift)&3];
 	memcpy(data, tmp, 4);
 }
-
+#endif
 static void aes_inv_shift_col(void* data, uint8_t shift){
 	uint8_t tmp[4];
 	tmp[0] = ((uint8_t*)data)[ 0];
diff --git a/drivers/amlogic/spi_nor/apollo_spi_nor.c b/drivers/amlogic/spi_nor/apollo_spi_nor.c
index b07c91187f2e..407a01119b4b 100755
--- a/drivers/amlogic/spi_nor/apollo_spi_nor.c
+++ b/drivers/amlogic/spi_nor/apollo_spi_nor.c
@@ -54,7 +54,7 @@ struct amlogic_spi {
 #endif
 };
 
-static bool spi_chip_select(bool flag);
+//static bool spi_chip_select(bool flag);
 //static DEFINE_SPINLOCK(pinmux_set_lock);
 
 #if (defined(CONFIG_ARCH_MESON6) || defined(CONFIG_ARCH_MESON8) || defined(CONFIG_ARCH_MESON8B))
@@ -114,12 +114,12 @@ static pinmux_set_t spi_nor_set = {
 };
 #endif
 
-
+#if 0
 static bool spi_chip_select(bool flag)
 {
 	return flag;
 }
-
+#endif
 static void spi_hw_init(struct amlogic_spi	*amlogic_spi)
 {
 #if  defined(ONFIG_AMLOGIC_BOARD_APOLLO) || defined(CONFIG_AMLOGIC_BOARD_APOLLO_H)
diff --git a/include/linux/thermal.h b/include/linux/thermal.h
index 78c3023fd7e8..098d49d220a0 100755
--- a/include/linux/thermal.h
+++ b/include/linux/thermal.h
@@ -261,16 +261,6 @@ static inline int thermal_generate_netlink_event(struct thermal_zone_device *tz,
 
 #ifdef CONFIG_AMLOGIC_THERMAL
 #include <mach/thermal.h>
-#else
-static int thermal_firmware_init(void)
-{
-	return -1;
-}
-
-static int get_cpu_temp(void)
-{
-	return -1;
-}
 #endif /* CONFIG_AMLOGIC_THERMAL */
 
 #endif /* __THERMAL_H__ */
diff --git a/sound/soc/aml/m6/aml_pcm.c b/sound/soc/aml/m6/aml_pcm.c
index cbfbb922e27a..c59c3d0eaefa 100755
--- a/sound/soc/aml/m6/aml_pcm.c
+++ b/sound/soc/aml/m6/aml_pcm.c
@@ -103,7 +103,7 @@ int codec_power=1;
 unsigned int flag=0;
 //static int num=0;
 
-static int codec_power_switch(struct snd_pcm_substream *substream, unsigned int status);
+//static int codec_power_switch(struct snd_pcm_substream *substream, unsigned int status);
 
 EXPORT_SYMBOL(aml_i2s_playback_start_addr);
 EXPORT_SYMBOL(aml_i2s_capture_start_addr);
@@ -112,7 +112,7 @@ EXPORT_SYMBOL(aml_i2s_capture_buf_size);
 EXPORT_SYMBOL(aml_i2s_playback_phy_start_addr);
 EXPORT_SYMBOL(aml_i2s_capture_phy_start_addr);
 EXPORT_SYMBOL(aml_i2s_alsa_write_addr);
-
+#if 0
 static void aml_codec_power_switch_queue(struct work_struct* work)
 {
 
@@ -122,7 +122,7 @@ static void aml_codec_power_switch_queue(struct work_struct* work)
 	// disable power down/up, which caused pop noise
 	//codec_power_switch(substream, clock_gating_status);
 }
-
+#endif
 /*--------------------------------------------------------------------------*\
  * Hardware definition
 \*--------------------------------------------------------------------------*/
@@ -467,6 +467,7 @@ static int audio_notify_hdmi_info(int audio_type, void *v){
 	audio_type_info = audio_type;
 
 }
+#if 0
 static void iec958_notify_hdmi_info(void)
 {
 	unsigned audio_type = AOUT_EVENT_IEC_60958_PCM;
@@ -487,6 +488,7 @@ static void iec958_notify_hdmi_info(void)
 	}
 
 }
+#endif
 /*
 special call by the audiodsp,add these code,as there are three cases for 958 s/pdif output
 1)NONE-PCM  raw output ,only available when ac3/dts audio,when raw output mode is selected by user.
@@ -855,6 +857,7 @@ static snd_pcm_uframes_t aml_pcm_pointer(
 	return 0;
 }
 
+#if USE_HRTIMER != 0
 static enum hrtimer_restart aml_pcm_hrtimer_callback(struct hrtimer* timer)
 {
   struct aml_runtime_data* prtd =  container_of(timer, struct aml_runtime_data, hrtimer);
@@ -902,7 +905,7 @@ static enum hrtimer_restart aml_pcm_hrtimer_callback(struct hrtimer* timer)
   hrtimer_forward_now(timer, ns_to_ktime(HRTIMER_PERIOD));
   return HRTIMER_RESTART;
 }
-
+#endif
 static void aml_pcm_timer_callback(unsigned long data)
 {
     struct snd_pcm_substream *substream = (struct snd_pcm_substream *)data;
@@ -1372,7 +1375,7 @@ static void aml_pcm_free_dma_buffers(struct snd_pcm *pcm)
     }
 }
 
-#ifdef CONFIG_PM
+#if 0
 static int aml_pcm_suspend(struct snd_soc_dai *dai)
 {
 	struct snd_pcm_runtime *runtime = dai->runtime;
@@ -1414,9 +1417,9 @@ static int aml_pcm_resume(struct snd_soc_dai *dai)
 
 #ifdef CONFIG_DEBUG_FS
 
-static struct dentry *debugfs_root;
-static struct dentry *debugfs_regs;
-static struct dentry *debugfs_mems;
+//static struct dentry *debugfs_root;
+//static struct dentry *debugfs_regs;
+//static struct dentry *debugfs_mems;
 
 static int regs_open_file(struct inode *inode, struct file *file)
 {
@@ -1608,7 +1611,7 @@ static const struct file_operations mems_fops={
 	.read = mems_read_file,
 	.write = mems_write_file,
 };
-
+#if 0
 static void aml_pcm_init_debugfs(void)
 {
 		debugfs_root = debugfs_create_dir("aml",NULL);
@@ -1631,6 +1634,7 @@ static void aml_pcm_cleanup_debugfs(void)
 {
 	debugfs_remove_recursive(debugfs_root);
 }
+#endif
 #else
 static void aml_pcm_init_debugfs(void)
 {
diff --git a/sound/soc/codecs/aml_pmu3.c b/sound/soc/codecs/aml_pmu3.c
index 512f6457bde7..bef6f2443d24 100755
--- a/sound/soc/codecs/aml_pmu3.c
+++ b/sound/soc/codecs/aml_pmu3.c
@@ -705,7 +705,7 @@ static int pmu3_hw_params(struct snd_pcm_substream *substream,
 	
 	return 0;
 }
-
+#if 0
 static int pmu3_set_dai_clkdiv(struct snd_soc_dai *codec_dai,
 		int div_id, int div)
 {
@@ -717,7 +717,7 @@ static int pmu3_set_dai_clkdiv(struct snd_soc_dai *codec_dai,
 
 	return 0;
 }
-
+#endif
 #define AML_PMU3_PLAYBACK_RATES SNDRV_PCM_RATE_8000_48000
 #define AML_PMU3_CAPTURE_RATES SNDRV_PCM_RATE_8000_48000
 #define AML_PMU3_FORMATS SNDRV_PCM_FMTBIT_S16_LE|SNDRV_PCM_FMTBIT_S24_LE
-- 
2.19.0

